packages/ember-views/lib/views/bound_component_view.js

```diff
@@ -0,0 +1,50 @@
+/**
+@module ember
+@submodule ember-views
+*/
+
+import { _Metamorph } from "ember-views/views/metamorph_view";
+import { read, chain, subscribe, unsubscribe } from "ember-metal/streams/utils";
+import { readComponentFactory } from "ember-views/streams/utils";
+import mergeViewBindings from "ember-htmlbars/system/merge-view-bindings";
+import EmberError from "ember-metal/error";
+
+export default Ember.ContainerView.extend(_Metamorph, {
```

Comment:
Since we only have a single child we do not need the power of `ContainerView` (to manage arrays of children). I think this can be done with a normal `Ember.View` (instead of a `Ember.ContainerView`).

I created a JSBin [here](http://emberjs.jsbin.com/rwjblue/186/edit?html,js,output) with a tiny demo (would still need the `_boundComponentProps` stuff you have. The basic gist is that adding a child view in `render` is cleared for every `rerender`.  You could extend `packages/ember-views/lib/mixins/normalized_rerender_if_needed.js` and set the `componentNameStream` at `lazyValue` (like [BoundPartialView](packages/ember-views/lib/views/bound_partial_view.js)).

@mmun / @mixonic - Does you agree?